arbiter/server.go

```diff
@@ -182,22 +188,26 @@ func (s *Server) Run() error {
 
 	var syncErr error
 
+	syncCtx, syncCancel := context.WithCancel(ctx)
 	wg.Add(1)
 	go func() {
 		defer wg.Done()
-		syncErr = syncBinlogs(s.kafkaReader.Messages(), s.load)
+		syncErr = syncBinlogs(syncCtx, s.kafkaReader.Messages(), s.load)
 		if syncErr != nil {
 			s.Close()
 		}
 	}()
 
 	err := s.load.Run()
 	if err != nil {
+		syncCancel()
 		s.Close()
```

Comment:
Maybe we can fix this problem in a simpler way? For example, it seems buggy that `Reader.Close` doesn't `close` the `Messages` channel.

Reply:
`Reader.Close` has closed successfully, but `Reader.msgs` channel has a buffer size of 1024. Func `syncCancel` can still read the buffered `msg` although it is closed.